### PR TITLE
MCHMP-1 Add autoconfiguration

### PR DESCRIPTION
**Pull request by persifal**
- externalize properties;
- add annotation processing maven;
- add properties metadata;
- update .gitignore;
- fix tests - types compatibility;
- formatting;
- add tests.

### CODE REVIEW

**Comment by YakovSirotkin** [comment]
@persifal I suppose we need to create `machamp-spring-boot-starter` module. We need to follow industry standards  to simplify learning curve for users.

File: machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt [hunk 1]
```diff
@@ -36,7 +29,7 @@ class AsyncTaskProcessor(
             taskHandlersMap[it.getType()] = it
         }
 
-        jobs = Array(threadsCount) { i ->
+        jobs = List(parallelismLevel) { i ->
```
Comment (YakovSirotkin) [inline machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt:32]: @persifal It's hard naming here, but parallelism level looks misleading: https://www.tutorialspoint.com/what-are-the-different-levels-of-parallelism. I see its usage in https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ForkJoinPool.html but it sounds like artefact from Russian engineers.

Comment (persifal) [reply machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt:32]: Yeah, agree the name might be not clear. Could you please suggest more meaningful variant to replace?

**Comment by persifal** [comment]
> @persifal I suppose we need to create machamp-spring-boot-starter module. We need to follow industry standards to simplify learning curve for users.


@YakovSirotkin  Completely agree. I was thinking of separate implementation from spring wrapper but realized that this task is out of this issue. I have [created issue](https://github.com/YakovSirotkin/machamp/issues/8) from you comment.

File: machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt (hunk 1 above)
Comment (YakovSirotkin) [reply machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt:32]: @persifal `threadsCount` is my best guess. Actually, this value not only about performance of this module, but about the load it generates to external services.

**Comment by YakovSirotkin** [comment]
> > @persifal I suppose we need to create machamp-spring-boot-starter module. We need to follow industry standards to simplify learning curve for users.
> 
> @YakovSirotkin Completely agree. I was thinking of separate implementation from spring wrapper but realized that this task is out of this issue. I have [created issue](https://github.com/YakovSirotkin/machamp/issues/8) from you comment.

That's my fault - I supposed machamp-spring-boot-starter in scope of this task, but failed to mention it in this issue.

File: machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt (hunk 1 above)
Comment (persifal) [reply machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt:32]: In this code we launch some coroutines which, in fact, are not threads.

**Comment by persifal** [comment]
> @persifal I suppose we need to create machamp-spring-boot-starter module. We need to follow industry standards to simplify learning curve for users.

>> @YakovSirotkin Completely agree. I was thinking of separate implementation from spring wrapper but realized that this task is out of this issue. I have https://github.com/YakovSirotkin/machamp/issues/8 from you comment.

>>> That's my fault - I supposed machamp-spring-boot-starter in scope of this task, but failed to mention it in this issue.

Time's passed :) lets separate it in the next iteration.

File: machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt (hunk 1 above)
Comment (YakovSirotkin) [reply machamp-core/src/main/kotlin/ru/telamon/machamp/AsyncTaskProcessor.kt:32]: > In this code we launch some coroutines which, in fact, are not threads.

Yes. But historically it was threads, before Kotlin development was started. And coroutine that loops infinitely and make pauses are like threads emulated with coroutines.

**Pull request closed without merging**